executor/src/witgen/jit/witgen_inference.rs

```diff
@@ -501,12 +542,12 @@ mod test {
         let mut complete = HashSet::new();
         let mut counter = 0;
         let expected_complete = expected_complete.unwrap_or(retained_identities.len() * rows.len());
-        while complete.len() != expected_complete {
+        while complete.len() < expected_complete {
```

Comment:
This change is not necessary, is it? We can keep it like that, just wondering if it was necessary.